src/opnsense/mvc/app/library/OPNsense/Backup/Base.php

```diff
@@ -85,10 +98,21 @@ public function decrypt($data, $pass, $tag = 'config.xml')
         @unlink("{$file}.dec");
 
         $data = explode("\n", $data);
+        $encryption_method = "M0"; /* default for backward compatibility */
 
         foreach ($data as $key => $val) {
-            /* XXX remove helper lines for now */
-            if (strpos($val, ':') !== false) {
+            if (strpos($val, "Encryption: ") === 0) {
+                /* honor encryption preamble field */
+                $encryption_method = explode(" ", $val, 2)[1];
+                if (!array_key_exists($encryption_method, $this->openssl_arguments)) {
```

Comment:
Security is ok by not being able to inject command line arguments but matching via magic constant is a bit counter-intuitive in general.

Reply:
Is it? Think of version numbers. As it's impossible to know how many and which parameters we might need in the future, using just a version-number like method scheme `M0`, `M1`, ... seems quite natural to me. The map helps to keep the code clean and simple, easy to extend and pretty self-documenting.

Reply:
Whats your opinion on using these labels as an alternative to the simple version-like scheme?
* `AES-256-CBC/MD5` instead of `M0`
* `AES-256-CBC/PBKDF2*100000/SHA512` instead of `M1`